sdk/storage/storage-blob/test/utils/index.ts

```diff
@@ -40,33 +38,49 @@ export function getGenericBSU(
     );
   }
 
-  const credentials = new SharedKeyCredential(accountName, accountKey);
-  const pipeline = newPipeline(credentials, {
+  return new SharedKeyCredential(accountName, accountKey);
+}
+
+export function getGenericBSU(
+  accountType: string,
+  accountNameSuffix: string = ""
+): BlobServiceClient {
+  const credential = getGenericCredential(accountType) as SharedKeyCredential;
+
+  const pipeline = newPipeline(credential, {
     // Enable logger when debugging
     // logger: new ConsoleHttpPipelineLogger(HttpPipelineLogLevel.INFO)
   });
-  const blobPrimaryURL = `https://${accountName}${accountNameSuffix}.blob.core.windows.net/`;
+  const blobPrimaryURL = `https://${credential.accountName}${accountNameSuffix}.blob.core.windows.net/`;
   return new BlobServiceClient(blobPrimaryURL, pipeline);
 }
 
+export function getTokenCredential(): TokenCredential {
+  const accountTokenEnvVar = `ACCOUNT_TOKEN`;
+  let accountToken: string | undefined;
+
+  accountToken = process.env[accountTokenEnvVar];
+
+  if (!accountToken || accountToken === "") {
+    throw new Error(`${accountTokenEnvVar} environment variables not specified.`);
+  }
+
+  return new SimpleTokenCredential(accountToken);
```

Comment:
As discussion with xiaoning, there could be more kinds of credentials involved in track 2, by identity package. For test coverage consideration, later suggest to add cases to cover them, e.g. in batch scenario.

Currently, batch has a case it("submitBatch should work with multiple types of credentials for subrequests") which covers anonymous, sharedkey, and token.

Reply:
Created an issue a few days ago to track the same(to test TokenCredentials from @azure/identity with storage SDKs(both node and browser) and to add some samples) - https://github.com/Azure/azure-sdk-for-js/issues/5051.

Reply:
That's good.